app/src/main/java/fr/gaulupeau/apps/Poche/events/EventProcessor.java

```diff
@@ -49,6 +51,45 @@ public void stop() {
         EventBus.getDefault().unregister(this);
     }
 
+    @Subscribe
+    public void onBootCompletedEvent(BootCompletedEvent event) {
+        Log.d(TAG, "onBootCompletedEvent() started");
+
+        Settings settings = getSettings();
+        if(settings.getBoolean(Settings.AUTOSYNC_ENABLED, false)) {
+            Log.d(TAG, "onBootCompletedEvent() setting an alarm");
+
+            AlarmHelper.setAlarm(getContext(),
+                    settings.getLong(Settings.AUTOSYNC_INTERVAL, AlarmManager.INTERVAL_DAY), false);
+        }
+    }
+
+    @Subscribe
+    public void onAlarmReceivedEvent(AlarmReceivedEvent event) {
+        Log.d(TAG, "onAlarmReceivedEvent() started");
+
+        Settings settings = getSettings();
+        if(!settings.getBoolean(Settings.AUTOSYNC_ENABLED, false)) {
+            Log.w(TAG, "onAlarmReceivedEvent() alarm received even though auto-sync is off");
+            return;
+        }
+
+        if(!WallabagConnection.isNetworkOnline()) {
+            Log.d(TAG, "No network, skipping auto-sync");
+            // TODO: set another closer alarm?
```

Comment:
I don't find it necessary. If I understand correctly, `networkChanged` will sync things anyway as soon as the network is on ?